src/views/News/index.js

```diff
@@ -19,530 +23,764 @@ import {
+  //Image isn't here because an image is optional
+  let content;
 
-  componentWillUnmount() {
-    window.removeEventListener('resize', this.resize);
-  }
+  if (props.authentication) {
+    if (loading === true) {
+      content = <GordonLoader />;
+    } else {
+      content = (
+        <NewsList
+          news={news}
+          personalUnapprovedNews={personalUnapprovedNews}
+          currentUsername={currentUsername}
+          updateSnackbar={updateSnackbar}
+          handleNewsItemEdit={handleNewsItemEdit}
+        />
+      );
+    }
 
-  render() {
-    // if all of the inputs are filled, enable 'submit' button
-    let submitButtonDisabled =
-      this.state.newPostCategory === '' ||
-      this.state.newPostSubject === '' ||
-      this.state.newPostBody === '';
-    let content;
-
-    /* Used to re-render the page when the network connection changes.
-     *  this.state.network is compared to the message received to prevent
-     *  multiple re-renders that creates extreme performance lost.
-     *  The origin of the message is checked to prevent cross-site scripting attacks
-     */
-    window.addEventListener('message', (event) => {
-      if (
-        event.data === 'online' &&
-        this.state.network === 'offline' &&
-        event.origin === window.location.origin
-      ) {
-        this.setState({ network: 'online' });
-      } else if (
-        event.data === 'offline' &&
-        this.state.network === 'online' &&
-        event.origin === window.location.origin
-      ) {
-        this.setState({ network: 'offline' });
-      }
-    });
-
-    /* Gets status of current network connection for online/offline rendering
-     *  Defaults to online in case of PWA not being possible
-     */
-    const networkStatus = JSON.parse(localStorage.getItem('network-status')) || 'online';
-
-    if (this.props.authentication) {
-      if (this.state.loading === true) {
-        content = <GordonLoader />;
-      } else if (this.state.news.length > 0 || this.state.personalUnapprovedNews.length > 0) {
-        content = (
-          <NewsList
-            news={this.state.filteredNews}
-            personalUnapprovedNews={this.state.personalUnapprovedNews}
-            updateSnackbar={this.updateSnackbar}
-            currentUsername={this.state.currentUsername}
-            callFunction={this.callFunction}
-          />
-        );
-      } else {
-        content = (
-          <Typography variant="h4" align="center">
-            No News To Show
-          </Typography>
-        );
-      }
+    let submitButton = (
+      <Button
+        variant="contained"
+        color="primary"
+        onClick={currentlyEditing ? handleUpdate : handleSubmit}
+        disabled={submitButtonDisabled}
+      >
+        {currentlyEditing ? 'Update' : 'Submit'}
+      </Button>
+    );
+
+    let newsJSX;
+
+    // If the user is online
+    if (isOnline || (!isOnline && props.authentication)) {
```

Comment:
The second case is never reached here because this is already inside an `if (props.authentication)` clause.